### PR TITLE
Add OpenSearch support for item searching

### LINKED ISSUE #355: Add support for OpenSearch

### PR DESCRIPTION
**Pull request by richardrodgers**
with Atom formatting.
Closes #355

### CODE REVIEW

File: app/controllers/Search.scala [hunk 1]
```diff
@@ -48,4 +56,92 @@ object Search extends Controller {
...
+      </OpenSearchDescription>
+    ).as("application/opensearchdescription+xml")
+  }
+
+  def itemResultsAtom(q: String, page: Int, perpage: Int, items: Seq[Item], total: Long)(implicit request: RequestHeader) = {
+    val lastPage = if (total.toInt % perpage > 0) (total.toInt / perpage) else (total.toInt / perpage) - 1
+    def schemeUrl(id: Int) = routes.Application.scheme(id).absoluteURL
+    def doiUrl(item: Item) = "http://doi.org/" + item.metadataValue("doi")
+    Ok(
+      <feed xmlns="http://www.w3.org/2005/Atom"
```
Comment (JPrevost) [inline app/controllers/Search.scala]: This document creation would be cleaner in a View.

File: app/controllers/Search.scala [hunk 2]
```diff
@@ -48,4 +56,92 @@ object Search extends Controller {
...
     }
   }
+
+  def openSearchDescription(dtype: String) = Action { implicit request =>
+    def templateUrl(ftype: String) = {
+      routes.Search.index.absoluteURL + "/results?q={searchTerms}&page={startPage?}&perpage={count?}" +
+      "&target=" + dtype + "&format=" + ftype
+    }
+    Ok(
+      <OpenSearchDescription xmlns="http://a9.com/-/spec/opensearch/1.1/">
```
Comment (JPrevost) [inline app/controllers/Search.scala]: This document creation would be cleaner in a View.

Comment (richardrodgers) [inline app/controllers/Search.scala]: Might be - our current practice (e.g. SWORD, Mets) for XML docs is not to use the view templates - should they be refactored as well?

Comment (JPrevost) [inline app/controllers/Search.scala]: In my opinion, ideally yes.

But honestly, the template generation is enough of a pain in the ass in Play that I'll definitely forgive you for choosing to leave this here. But keep in mind I just might use this as an excuse for generating HTML in Controllers and dropping Twirl entirely ;) [Don't worry, that won't happen even if I kinda want to]. I'd suggest we leave this here for now and refactor all the generated XML to templates if it makes sense later.

Comment (richardrodgers) [inline app/controllers/Search.scala]: I think it's a tough call, and there are multiple issues in play (no pun intended) here:

First, there is the fact that scala has built-in affordances for XML manipulation that it doesn't for HTML, etc that makes XML wrangling easy in code. Not clear there's a win in readability, etc to push XML into a Twirl template - maybe the reverse.

Second, there is the design issue of clean separation of concerns that argues for separating controller and view code, and this pulls the other way - towards moving things into view-land.

Third, there is the question of strengths and weaknesses of Twirl as a template engine itself, where I think arguments can be made either way.

I think we can look at this more systematically when we build an API, where the notion of a 'view' becomes more general than a web page/XML document. We might want a notion of view that encompasses more the twirl, etc 

But all that's more than I wanted to tackle with this small feature....

Comment (JPrevost) [inline app/controllers/Search.scala]: Agree on all points.

**Comment by JPrevost** [comment]
:+1:

**Pull request merged**